third_party/move/move-model/bytecode/src/function_target_pipeline.rs

```diff
@@ -385,45 +392,54 @@ impl FunctionTargetPipeline {
         let rev_topo_order = Self::sort_in_reverse_topological_order(env, targets);
         info!("transforming bytecode");
         hook_before_pipeline(targets);
-        for (step_count, processor) in self.processors.iter().enumerate() {
-            if processor.is_single_run() {
-                processor.run(env, targets);
-            } else {
-                processor.initialize(env, targets);
-                for item in &rev_topo_order {
-                    match item {
-                        Either::Left(fid) => {
-                            let func_env = env.get_function(*fid);
-                            targets.process(&func_env, processor.as_ref(), None);
-                        },
-                        Either::Right(scc) => 'fixedpoint: loop {
-                            let scc_env: Vec<_> =
-                                scc.iter().map(|fid| env.get_function(*fid)).collect();
-                            for fid in scc {
-                                let func_env = env.get_function(*fid);
-                                targets.process(&func_env, processor.as_ref(), Some(&scc_env));
-                            }
-
-                            // check for fixedpoint in summaries
-                            for fid in scc {
+        let mut changed = true;
+        while changed {
```

Comment:
You seem to redesign the semantics of this central function which is used in 1/2 dozen of tools in this repo. Apparently, if you have a given pipeline, it will now repeatedly run until code does not change any more. This requires processors to be idempotent, beside other issues like unnecessarily running the pipeline twice. We can't do this. If you want a fixed point loop over a sequence of processors, please implement this in the decompiler. You could, for example, attach an annotation `CodeChangedAnnotation(bool)`and read it by an outer loop.